docs/source/quickstart.mdx

```diff
@@ -19,13 +19,14 @@ The following code illustrates the steps above.
 # 0. imports
 import torch
 from transformers import GPT2Tokenizer
-from trl import PPOTrainer, PPOConfig, AutoModelForCausalLMWithValueHead, create_reference_model
+from trl import PPOTrainer, PPOConfig, AutoModelForCausalLMWithValueHead
 from trl.core import respond_to_batch
```

Comment:
we should also replace `respond_to_batch` with `ppo_trainer.generate`. could you test if that works?

Reply:
Thanks for the comment. I replaced it with `ppo_trainer.generate` as follows, but I have to use `query_tensor[0]` instead of `query_tensor`. 

```python
response_tensor = ppo_trainer.generate(query_tensor[0], return_prompt=False)
response_txt = tokenizer.decode(response_tensor[0,:])
```

Furthermore it gives two extra warnings:

```
The attention mask and the pad token id were not set. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
Setting `pad_token_id` to `eos_token_id`:50256 for open-end generation.
/admin/home/costa/.pyenv/versions/trl/lib/python3.8/site-packages/transformers/generation/utils.py:1346: UserWarning: Using `max_length`'s default (20) to control the generation length. This behaviour is deprecated and will be removed from the config in v5 of Transformers -- we recommend using `max_new_tokens` to control the maximum length of the generation.
```


I then followed the example in sentiment analysis to do the following

```
generation_kwargs = {
    "min_length": -1,
    "top_k": 0.0,
    "top_p": 1.0,
    "do_sample": True,
    "pad_token_id": tokenizer.eos_token_id,
}
output_min_length = 4
output_max_length = 16
output_length_sampler = LengthSampler(output_min_length, output_max_length)
response_tensor = ppo_trainer.generate([item for item in query_tensor], return_prompt=False, length_sampler=output_length_sampler, **generation_kwargs)
response_txt = tokenizer.decode(response_tensor[0])
```

which removes the warning. 

Please let me know which one do you prefer :)

Reply:
Oh yes, you can add `max_new_tokens=10` (or whatever number) to the generation kwargs.